### PR TITLE
Issue#86 - Test `NewQbDimension.from_data

### PR DESCRIPTION
**Pull request by santhosh-thangavel**
1. Test case is added to check if NewQbDimension correctly extracts and creates a Valid NewQbCodeList & NewQbConcept
2.Unique concept labels are asserted
3.Asserted NewQbDimension concepts does not have a parent
4. Asserted to see if concept labels are converted into something that can be used in a URI path segment (pathify).

#86

### CODE REVIEW

**Comment by santhosh-thangavel** [comment]
@canwaf Please, review my pull request.

File: csvqb/csvqb/tests/unit/cube/qb/test_qb_dimension.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    # List of concept labels are asserted Individually
+    concept_labels = [c.label for c in new_dimension.code_list.concepts]
+    assert "A" in concept_labels
+    assert "C" in concept_labels
+    assert "D" in concept_labels
+    assert "E" in concept_labels
+    assert "G" in concept_labels
+
+    # assert parent_code is None
+    assert new_dimension.code_list.concepts[0].parent_code is None
```
Comment (robons) [inline csvqb/csvqb/tests/unit/cube/qb/test_qb_dimension.py:34]: It would be a good idea if we could assert this for *all* concepts in the codelist.

Comment (santhosh-thangavel) [reply csvqb/csvqb/tests/unit/cube/qb/test_qb_dimension.py]: asserted parent is None for list of distinct concepts.

File: csvqb/csvqb/tests/unit/cube/qb/test_qb_dimension.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    concept_labels = [c.label for c in new_dimension.code_list.concepts]
+    assert "A" in concept_labels
+    assert "C" in concept_labels
+    assert "D" in concept_labels
+    assert "E" in concept_labels
+    assert "G" in concept_labels
+
+    # assert parent is None for distinct list of concepts
+    parent_code_var = [c.parent_code for c in new_dimension.code_list.concepts]
+    assert parent_code_var[0] is None
```
Comment (robons) [inline csvqb/csvqb/tests/unit/cube/qb/test_qb_dimension.py:35]: There's a nicer way to do this using the [all](https://docs.python.org/3/library/functions.html#all) function with a list comprehension.

Want to give that a go?

Comment (santhosh-thangavel) [reply csvqb/csvqb/tests/unit/cube/qb/test_qb_dimension.py:35]: Sure

**Comment by santhosh-thangavel** [comment]
@canwaf Please, review my pull request.

File: csvqb/csvqb/tests/unit/cube/qb/test_qb_dimension.py (hunk 2 above)
Comment (santhosh-thangavel) [reply csvqb/csvqb/tests/unit/cube/qb/test_qb_dimension.py]: Used [any](https://www.w3.org/2009/08/skos-reference/skos.html#hasTopConcept) function with a list comprehension

**Review by canwaf — APPROVED** [review]

**Pull request merged**